gcloud-java-core/src/main/java/com/google/gcloud/SigningAuthCredentials.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+/**
+ * Interface for authentication credentials that support signing.
```

Comment:
Do you really think we should mention here (Javadoc, name,..) that it is intended only for Credentials?

I am not convinced, and therefore would have called it with a name more generic (e.g. `Signer` or `ServiceAccountSigner`) and document it as such (after all one could provide an instance of it for the signUrl
params and without supporting all the other auth credentials requirements). 

Otherwise, any reason for not making it a specific subclass of `AuthCredentials`?

My preference would be the former.